### PR TITLE
Update context menu items for attached storage accounts

### PR DESCRIPTION
**Pull request by wwlorey**
Fixes https://github.com/microsoft/vscode-azurestorage/issues/636

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -449,7 +449,7 @@
 				},
 				{
 					"command": "azureStorage.createBlockBlob",
-					"when": "view == azureStorage && viewItem == azureBlobContainer",
+					"when": "view == azureStorage && viewItem == azureBlobContainer || viewItem == azureBlobContainer-attached",
```
Comment (nturinski) [inline package.json]: Why not just use the regexp for this?

File: package.json [hunk 2]
```diff
@@ -484,22 +484,22 @@
...
 					"command": "azureStorage.refresh",
-					"when": "view == azureStorage && viewItem == azureBlobContainer",
+					"when": "view == azureStorage && viewItem == azureBlobContainer || viewItem == azureBlobContainer-attached",
 					"group": "9_refresh"
 				},
 				{
 					"$comment": "========= azureBlob =========",
 					"command": "azureStorage.deleteBlob",
-					"when": "view == azureStorage && viewItem == azureBlob"
+					"when": "view == azureStorage && viewItem == azureBlob || viewItem == azureBlob-attached"
```
Comment (nturinski) [inline package.json]: Same with these

**Comment by EricJizbaMSFT** [comment]
Yes I agree with Nathan. The amount of changes are not worth the benefit it provides

File: package.json (hunk 1 above)
Comment (wwlorey) [reply package.json]: I'd have to exclude the matches on `azureBlobContainerGroup` so I decided to explicitly check

**Comment by wwlorey** [comment]
I'd be fine with putting it off. It is nice to only have the context menu options that apply but it's a lot of changes

File: src/tree/AttachedStorageAccountsTreeItem.ts [hunk 3]
```diff
@@ -65,7 +63,7 @@ export class AttachedStorageAccountsTreeItem extends AzureParentTreeItem {
     }
 
     public isAncestorOfImpl(contextValue: string): boolean {
-        return contextValue !== StorageAccountTreeItem.contextValue;
+        return !!contextValue.match(/-attached/);
```
Comment (wwlorey) [inline src/tree/AttachedStorageAccountsTreeItem.ts]: Just wanted to bring this up so everybody's aware of it:

Without this change to `isAncestorOfImpl`, you can choose an attached or emulated account for commands like `Create Storage Account`

Screen Shot 2020-03-12 at 4 16 42 PM

And that leads to an error like this

Screen Shot 2020-03-12 at 4 18 10 PM

I don't think that error is a dealbreaker but worth mentioning

Comment (EricJizbaMSFT) [reply src/tree/AttachedStorageAccountsTreeItem.ts]: I feel like you could probably do a one-line fix as a separate PR to fix that

Comment (EricJizbaMSFT) [reply src/tree/AttachedStorageAccountsTreeItem.ts]: How about this:
```
return contextValue !== StorageAccountTreeItem.contextValue && contextValue !== SubscriptionTreeItem.contextValue;
```

Comment (wwlorey) [reply src/tree/AttachedStorageAccountsTreeItem.ts]: Ah that fixes it!

**Comment by wwlorey** [comment]
Closing since the benefit this provides isn't worth this many changes

**Pull request closed without merging**